src/plugins/hetero/executable_network.cpp

```diff
@@ -66,8 +67,13 @@ HeteroExecutableNetwork::HeteroExecutableNetwork(const InferenceEngine::CNNNetwo
       _heteroPlugin{plugin},
       _name{network.getName()},
       _config{config} {
+
     auto function = network.getFunction();
     IE_ASSERT(function != nullptr);
+
+    _manager.register_pass<ngraph::pass::ConstantFolding>();
+    _manager.run_passes(std::const_pointer_cast<ov::Model>(function));
```

Comment:
do we really need to put device specific code to common logic on hetero?

Reply:
Hi

As I understand, hetero plugin may split network to subgraphs with dynamic parts (split shapeOf and following operations). That leads some dynamic parts may be assigned to devices that does not support dynamic networks. Constant folding in hetero plugin works around this issue until solution is implemented

Reply:
It is ok, that we apply constant folding. It is a WA to fuse constant branches